core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -349,9 +360,14 @@ object DumpLogSegments {
               " headerKeys: " + record.headers.map(_.key).mkString("[", ",", "]"))
           }
 
-          if (record.isControlRecord) {
+          if (batch.isControlBatch) {
             val controlType = ControlRecordType.parse(record.key)
             print(s" controlType: $controlType")
+
+            if (controlType == ControlRecordType.ABORT || controlType == ControlRecordType.COMMIT) {
+              val endTxnMarker = EndTransactionMarker.deserialize(record)
+              print(s" coordinatorEpoch: ${endTxnMarker.coordinatorEpoch}")
+            }
```

Comment:
I added some code to print the unknown type id in the else case. Let me know if that seems reasonable.